### DISCUSSION TITLE
Lenia Computing Power limitation

### DISCUSSION THREAD

**Discussion by Takodachi6969** [Part-II-Computational-Physics/.github#95] [General]
I am doing the project on Lenia and I have decided to find relationships between different life forms' shape features and their behaviours. A very promising result was made with a dataset of 10 (Which took the entire afternoon to run). However, when I try to scale up using more data (like 300 of them), the notebook just won't do it and I am kicked out of run time after running for a while.
Are there any suggestions for letting it run overnight or faster? I have tried optimising the code, but i doubt there are too much to be done with these.

if I cannot scale it up, will I lose marks? Here is a part of the result, but since there are only 10 data points there will be so much uncertainty its kind of meaningless. 
<img width="492" alt="Screenshot 2023-04-11 175841" src="https://user-images.githubusercontent.com/102265625/231127312-3f6e846f-9985-4398-8e50-f7c9e65af94c.png">

**Comment by AustenLamacraft** [comment]
I guess one thing to do would be to understand what determines the run time here. At the very least you can discuss what the various factors are, and what would be required to scale it up

**Reply by Takodachi6969** [reply]
There was an issue with the code that caused it. Fixed, thank you!

**Comment by jiangzhenyao** [comment]
Hi! I am also doing Lenia, I wonder where you get those data of different life forms from?

**Reply by Takodachi6969** [reply]
They are from the Lenia GitHub page, it was hidden as a Json file somewhere that you need to decode to use

**Reply by jiangzhenyao** [reply]
Thank you for your reply! I have found the Json file. Can I ask you how did you decode the file? I have tried UFT-8 and the convert.py file, but none of them works. I have been working on decoding the file the whole morning and I can not figure out how.

**Reply by Takodachi6969** [reply]
Well, you do need to use UFT-8 since there are Taiwanese in the file, the array was decoded using rle, which was a standard encoding method for cellular automata (he said it is standard in the paper but apparently only he uses it)..
